samples/cpp/text2image/heterogeneous_stable_diffusion.cpp

```diff
@@ -44,32 +44,47 @@ int32_t main(int32_t argc, char* argv[]) try {
     // Note that we could have created the scheduler by specifying specific type (for example EULER_DISCRETE), like
     // this: auto scheduler = ov::genai::Scheduler::from_config(root_dir / "scheduler/scheduler_config.json",
     //                                                    ov::genai::Scheduler::Type::EULER_DISCRETE);
+    // This can be useful when a particular type of Scheduler is not yet supported natively by OpenVINO GenAI.
+    // (even though we are actively working to support most commonly used ones)
 
     // Create unet object
     auto unet = ov::genai::UNet2DConditionModel(root_dir / "unet");
 
-    // Given the guidance scale, etc., calculate the batch size.
-    int unet_batch_size = 1;
-    if (guidance_scale > 1.0f && unet.get_config().time_cond_proj_dim < 0) {
-        unet_batch_size = 2;
-    }
+    // Set batch size based on classifier free guidance condition.
+    int unet_batch_size = unet.do_classifier_free_guidance(guidance_scale)  ? 2 : 1;
```

Comment:
for Python sample, you would also need to add pybind11 wrapper for this metho here https://github.com/openvinotoolkit/openvino.genai/blob/master/src/python/py_image_generation_models.cpp#L93